src/main/java/com/kryptnostic/api/v1/sharing/IndexPair.java

```diff
@@ -0,0 +1,29 @@
+package com.kryptnostic.api.v1.sharing;
+
+import com.kryptnostic.krypto.engine.KryptnosticEngine;
+
+public final class IndexPair {
+    private IndexPair( byte[] objectSearchKey, byte[] objectAddressMatrix ) {
+        this.objectSearchKey = objectSearchKey;
+        this.objectAddressMatrix = objectAddressMatrix;
+    }
+
+    private final byte[] objectSearchKey;
+    private final byte[] objectAddressMatrix;
+
+    public static IndexPair newFromKryptnosticEngine( KryptnosticEngine engine ) {
+        return new IndexPair( engine.getObjectSearchKey(), engine.getObjectAddressMatrix() );
+    }
+
+    public byte[] getObjectSearchKey() {
+        return objectSearchKey;
+    }
+
+    public byte[] getObjectAddressMatrix() {
+        return objectAddressMatrix;
+    }
+    
+    public byte[] computeSharingPair( KryptnosticEngine engine ) {
```

Comment:
Correct, this is a work around for now. We're going to need to re-work the C++ code to allow computing a sharing pair off individual elements.